File: rudolph-annotations/src/main/java/cn/wzbos/android/rudolph/annotations/Export.java
package cn.wzbos.android.rudolph.annotations;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 标记当前路由需要导出到协议层
 */
@Documented
@Retention(RetentionPolicy.CLASS)
@Target({ElementType.TYPE})
public @interface Export {
    /**
     * 指定导出到协议层类名
     */
    String value() default "";
}

File: rudolph-annotations/src/main/java/cn/wzbos/android/rudolph/annotations/Export.kt
package cn.wzbos.android.rudolph.annotations

/**
 * 标记当前路由需要导出到协议层
 */
@MustBeDocumented
@kotlin.annotation.Retention(AnnotationRetention.BINARY)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class Export(
    /**
     * 指定导出到协议层类名
     */
    val value: String = ""
)